kyo-core/shared/src/main/scala/kyo/Queue.scala

```diff
@@ -372,6 +388,18 @@ object Queue:
                     )
                 def poll()(using AllowUnsafe) = op(Maybe(q.poll()))
                 def peek()(using AllowUnsafe) = op(Maybe(q.peek()))
+                protected def _drainUpTo(max: Int): Seq[A] =
+                    val b = Seq.newBuilder[A]
+                    @tailrec def loop(i: Int): Unit =
+                        if i < max then
+                            val value = q.poll()
+                            if !isNull(value) then
+                                b.addOne(value)
+                                loop(i + 1)
+                    end loop
+                    loop(0)
+                    b.result()
+                end _drainUpTo
                 def _drain() =
```

Comment:
It'd be nice to avoid the code duplication. It seems `drainUpTo` can be implemented in terms of `drain` with a max `Int.MaxValue`?